app/src/main/java/com/example/androidtrainingproject/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import coil.compose.AsyncImage
+import com.example.androidtrainingproject.R
+import com.example.androidtrainingproject.ui.shared.AppTopBar
+import com.example.androidtrainingproject.ui.shared.SearchBox
+import com.example.androidtrainingproject.ui.theme.Gray
+import com.example.androidtrainingproject.ui.theme.LabelText
+import com.example.androidtrainingproject.ui.theme.Purple
+
+@Composable
+fun Home(navController: NavController) {
+    val homeViewModel: HomeViewModel = hiltViewModel()
+    val searchText by homeViewModel.searchText.collectAsState()
+
+    Scaffold(
+        topBar = {
+            AppTopBar(
+                middleText =  stringResource(id = R.string.home_top_bar_text),
+                navController = navController,
+                accountIcon = true,
+                cartIcon = true
+            )
+        }
+    ) { paddingValues ->
+        Column(
+            Modifier
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .paint(
+                    painter = painterResource(id = R.drawable.app_background),
+                    contentScale = ContentScale.FillBounds
+                )
+                .padding(top = paddingValues.calculateTopPadding())
+                .padding(dimensionResource(id = R.dimen.padding_m))
+        ) {
+            SearchBox(
+                value = searchText,
+                onChange = { value -> homeViewModel.updateSearchText(value) },
+                clear = { homeViewModel.clearSearchText() }
+            )
+            Row(
+                modifier = Modifier.padding(top = dimensionResource(id = R.dimen.padding_m)),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = stringResource(id = R.string.home_new_products),
+                    fontSize = dimensionResource(id = R.dimen.font_size_l).value.sp
+                )
+                Spacer(modifier = Modifier.weight(0.5f))
+                Icon(
+                    painter = painterResource(id = R.drawable.ic_filter_list),
+                    contentDescription = null,
+                    modifier = Modifier.height(dimensionResource(id = R.dimen.icon_large))
+                )
+            }
+
+            if (homeViewModel.isLoading) {
+                Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
+                    CircularProgressIndicator()
+                }
+            }
+
+            homeViewModel.filteredProducts.let { products ->
+                LazyColumn(modifier = Modifier
+                    .fillMaxWidth()
+                    .weight(1f)) {
```

Comment:
Doesn't `.fillMaxSize()` achieve what you strive for here instead of this `weight`?